flake.nix

```diff
@@ -1,95 +1,181 @@
+
+        fixPngOptimization = pkgs.callPackage ./nix/fix-png-optimization { };
+        fixStylishHaskell = pkgs.callPackage ./nix/fix-stylish-haskell { };
+
+        writeShellScriptBinInRepoRoot = name: script: pkgs.writeShellScriptBin name ''
+          cd `${pkgs.git}/bin/git rev-parse --show-toplevel`
+          ${script}
+        '';
+
+        scripts = import ./nix/scripts.nix {
+          inherit pkgs easyPS writeShellScriptBinInRepoRoot;
+          inherit (pkgs.nodePackages) prettier;
+
         };
-      in
-      {
-        packages = topLevel.bitte-packages // {
-          entrypoints = topLevel.bitte-packages;
+
+        formatting = import ./nix/formatting.nix {
+          inherit writeShellScriptBinInRepoRoot pkgs easyPS;
+        };
+
+        tests = import ./nix/tests/default.nix {
+          inherit pkgs fixPngOptimization fixStylishHaskell;
+          inherit (formatting) fix-prettier;
+          inherit (easyPS) purs-tidy;
+          src = ./.;
+          play-generated = scripts.generated-purescript;
+        };
+        pre-commit-check = pre-commit-hooks.lib.${system}.run {
+          src = (pkgs.lib.cleanSource ./.);
+          hooks = {
+            prettier = {
+              enable = true;
+              types_or = [ "javascript" "css" "html" ];
+            };
+            purs-tidy-hook = {
+              enable = true;
+              name = "purs-tidy";
+              description = "Ensure PureScript files are formatted";
+              entry = "${easyPS.purs-tidy}/bin/purs-tidy format-in-place";
+              files = "\\.purs$";
+              language = "system";
+            };
+            shellcheck.enable = true;
+            stylish-haskell.enable = true;
+            nixpkgs-fmt.enable = true;
+            png-optimization = {
+              enable = true;
+              name = "png-optimization";
+              description = "Ensure that PNG files are optimized";
+              entry = "${pkgs.optipng}/bin/optipng";
+              files = "\\.png$";
+            };
+          };
         };
-      }));
+        overlays = [
+          haskellNix.overlay
+          iohkNix.overlays.crypto
+          (final: prev: {
+            playground =
+              final.haskell-nix.cabalProject' {
+                src = ./.;
+                compiler-nix-name = "ghc8107";
+                modules = [
+                  {
+                    packages.plutus-script-utils.ghcOptions = [ "-Wwarn" "-Wno-unused-packages" ];
+                    # See https://github.com/input-output-hk/iohk-nix/pull/488
+                    packages.cardano-crypto-praos.components.library.pkgconfig = pkgs.lib.mkForce [ [ pkgs.libsodium-vrf pkgs.secp256k1 ] ];
+                    packages.cardano-crypto-class.components.library.pkgconfig = pkgs.lib.mkForce [ [ pkgs.libsodium-vrf pkgs.secp256k1 ] ];
+                  }
+                ];
+                shell.tools = {
+                  cabal = { };
+                  # hlint = {};
```

Comment:
How come this is commented out?

Reply:
We are not using it (for the moment)... removing it